shelley/chain-and-ledger/executable-spec/test/Rules/TestDeleg.hs

```diff
@@ -79,3 +80,42 @@ rewardZeroAfterReg = withTests (fromIntegral numberOfTests) . property $ do
           (   hk ∈ getStDelegs d'
            && (Maybe.maybe True (== 0) $ Map.lookup (mkRwdAcnt hk) (getRewards d')))
         credNewlyRegisteredAndRewardZero (_, _, _) = True
+
+-- | Check that when a stake credential is deregistered, it will not be in the
+-- rewards mapping or delegation mapping of the target state.
+credentialRemovedAfterDereg :: Property
+credentialRemovedAfterDereg = withTests (fromIntegral numberOfTests) . property $ do
+  t <- forAll (trace @DELEG $ fromIntegral traceLen)
+  let
+    n :: Integer
+    n = fromIntegral $ traceLength t
+    tr = sourceSignalTargets t
+
+  when (n > 1) $
+    True === (all removedDeregCredential tr)
```

Comment:
Maybe use `assert`?

Reply:
I'm also thinking wouldn't it make sense to check that the set of non-removed delegation credential is empty? The difference here is that if the tests fail we could see which credentials didn't get removed.